src/schema/feeds.ts

```diff
@@ -935,7 +935,11 @@ const legacySimilarPostsResolver = randomPostsResolver(
 export const resolvers: IResolvers<any, Context> = traceResolvers({
   Query: {
     anonymousFeed: (source, args: AnonymousFeedArgs, ctx: Context, info) => {
-      if (args.version >= 2 && args.ranking === Ranking.POPULARITY) {
+      if (
+        args.version >= 2 &&
+        args.ranking === Ranking.POPULARITY &&
+        ctx.userId
```

Comment:
But this is already in annonymousfeed no?
Also think we dknt really surface these anymore in favor of feedservice (but could have some use still)

Reply:
It's used for the popular feed. it's possible that some users have an old version. I think it's a valid solution as a quick hack for now

Reply:
Probably don't need it actually, the other fix should already take care of the rouge query.

Reply:
Ideally it should be controlled on entry level eg. routes that need or do not need auth, but as a quick fix to confirm it works I agree its fine.